src/Microsoft.Health.Fhir.Shared.Core/Features/Operations/Everything/PatientEverythingService.cs

```diff
@@ -0,0 +1,369 @@
+                    {
+                        token.Phase = 2;
+                        goto case 2;
+                    }
+
+                    break;
+                case 2:
+                    searchResult = start == null && end == null
+                        ? await SearchCompartment(resourceType, resourceId, since, type, encodedInternalContinuationToken, cancellationToken)
+                        : await SearchCompartmentWithoutDate(resourceType, resourceId, since, type, encodedInternalContinuationToken, cancellationToken);
+
+                    if (!searchResult.Results.Any())
+                    {
+                        token.Phase = 3;
+                        goto case 3;
+                    }
+
+                    break;
+                case 3:
+                    searchResult = await SearchRevinclude(resourceId, since, type, encodedInternalContinuationToken, cancellationToken);
+                    break;
+                default:
+                    return new SearchResult(Enumerable.Empty<SearchResultEntry>(), null, null, Array.Empty<Tuple<string, string>>());
+            }
+
+            string newContinuationToken = null;
+            if (searchResult.ContinuationToken != null)
+            {
+                newContinuationToken = EverythingOperationContinuationToken.ToString(token.Phase, searchResult.ContinuationToken);
+            }
+            else if (token.Phase < 3)
+            {
+                newContinuationToken = EverythingOperationContinuationToken.ToString(token.Phase + 1, null);
+            }
+
+            return new SearchResult(searchResult.Results, newContinuationToken, searchResult.SortOrder, searchResult.UnsupportedSearchParameters);
+        }
+
+        private async Task<SearchResult> SearchIncludes(
+            string resourceType,
+            string resourceId,
+            PartialDateTime since,
+            string type,
+            CancellationToken cancellationToken)
+        {
+            using IScoped<ISearchService> search = _searchServiceFactory();
+            var searchResultEntries = new List<SearchResultEntry>();
+
+            // Build search parameters
+            var searchParameters = new List<Tuple<string, string>>
+            {
+                Tuple.Create(SearchParameterNames.Id, resourceId),
+            };
+
+            searchParameters.AddRange(_includes.Select(include => Tuple.Create(SearchParameterNames.Include, $"{resourceType}:{include}")));
+
+            // Search
+            SearchOptions searchOptions = _searchOptionsFactory.Create(resourceType, searchParameters);
+            SearchResult searchResult = await search.Value.SearchAsync(searchOptions, cancellationToken);
+            searchResultEntries.AddRange(searchResult.Results.Select(x => new SearchResultEntry(x.Resource)));
+
+            // Filter results by IsDeleted
+            searchResultEntries = searchResultEntries.Where(s => !s.Resource.IsDeleted).ToList();
```

Comment:
Are we actually return deleted resources?
I don't this should be the case.

Reply:
`SearchAsync` in Cosmos DB does not return deleted resources. But we when pulling `_include`, we use `_fhirDataStore.GetAsync` and this does not limit `isDeleted=false`. So a deleted resource will be returned, but only contains some metadata. Here we just filter it out to keep consistency with other resources pulled by `SearchAsync`.

Reply:
That sounds like a bug, on our side...

Reply:
Agree that it might be better to remove deleted resources in `FhirCosmosSearchService.PerformIncludeQueries()`, though those deleted resources only contain some metadata.

Reply:
Since PR #1956 fixed this, the isDeleted check is removed in the post-process of $everything operation.